docs/arch/adr-33.md

```diff
@@ -0,0 +1,29 @@
+# ADR 33: Survey Task Feedback
+
+July 21, 2021
+
+## Context
+
+We plan to implement feedback for the survey task.
+
+Currently in PFE, the only feedback related to the survey task is for the Gravity Spy project, and is referred to as "Gravity Spy Gold Standard". Gravity Spy Gold Standard is a PFE experimental feature (per workflow configuration object) that checks a subject's metadata for `'#Type' === 'Gold'` and provides feedback after classification completion with the [`gs-gold-standard-summary`](https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/classifier/gs-gold-standard-summary.jsx). Survey task annotation values are compared to subject metadata `'#Label'` values. A message is then shown accordingly (i.e. success, failure, or special message per subject metadata `'#post_classification_feedback'`).
+
+## Decision
+
+To implement a survey task feedback strategy within [the existing feedback framework](https://github.com/zooniverse/front-end-monorepo/tree/master/packages/lib-classifier/src/store/feedback/strategies). The initial survey task annotation feedback will presume the annotation has one annotation value per survey task choice. The subject metadata will include:
+
+- #feedback_N_id (required) - ID of the corresponding workflow task rule.
+- #feedback_N_choice_labels (required) - comma separated target choice label(s) (not values, i.e. "Blip" not "BLP").
+- #feedback_N_choice_strict (optional) - defaults to true, target label(s) must match annotation label(s) exactly for success. If false then any target label(s) match with annotation label(s) will be a success.
```

Comment:
I'm not sure about this, curious what others think. Can leave for subsequent iteration and just keep strict match initially.

Reply:
I don't know how we would do a fuzzy match without an algorithm. Did you have something in mind? This implies we have the capability to do this (or will have the capability), and if we don't, then I don't think we should give people the option.

Reply:
This would be if the subject metadata feedback says the answer is 'BLIP' and the user has answered 'BLIP' and 'KOIFISH', or vice versa (feedback says two choices, user has answered with one) is the classification a success or failure. I think if strict is true [this line stays as is](https://github.com/zooniverse/Panoptes-Front-End/blob/master/app/classifier/gs-gold-standard-summary.jsx#L18), if strict is false it would use `Array.some`, I think, though likely oversimplifying in my mind. I think that's possible without an algorithm or additional dependency, but could be misunderstanding?

For Gravity Spy it'd be strict = true, but maybe camera trap projects would want strict = false? This can 💯  be a subsequent iteration if it's even something camera trap researchers want.

Reply:
I misunderstood what this meant. I thought it was referring to the string comparisons itself which would require fuzzy string matching (basically an algorithm). Array.every vs Array.some is ok though. We just need to make sure this is clearly documented about the array or expected list. Thanks!

Reply:
Could we add to the docs here that true means array.every and false means array.some?

Reply:
Apologies for the noise around this, but in a similar mindset that the initial focus is a simple survey task feedback strategy, and subsequently we can create more complex survey task feedback strategies that include question answers, I think this "strict" attribute should be dropped, and only exact matches should be a success?

Then based on input from researchers if it's something desired we can revisit, it wouldn't even have to be a new strategy type, but could be an iteration on the simple survey task feedback strategy if the default is going to be a strict comparison. Or it could be part of a new complex type. Important point I think is that it's not necessary initially?

I'll update accordingly, but happy to still include if others feel strongly it should be included.

Reply:
Sounds good to me. This could be in a future iteration if asked for.